admin/modules/config/settings.php

```diff
@@ -958,18 +958,51 @@
-		   || (in_array($mybb->settings['captchaimage'], array(8)) && (!$mybb->settings['recaptchaprivatekey'] || !$mybb->settings['recaptchapublickey'] || !$mybb->settings['recaptchascore']))
-		   || (isset($mybb->input['upsetting']['captchaimage']) && in_array($mybb->input['upsetting']['captchaimage'], array(6, 7)) && (!$mybb->input['upsetting']['hcaptchaprivatekey'] || !$mybb->input['upsetting']['hcaptchapublickey']))
-		   || (in_array($mybb->settings['captchaimage'], array(6, 7)) && (!$mybb->settings['hcaptchaprivatekey'] || !$mybb->settings['hcaptchapublickey'])))
-		{
-			$mybb->input['upsetting']['captchaimage'] = 1;
+		require_once MYBB_ROOT.'inc/class_captcha.php';
+		
+		// Have we opted for a reCAPTCHA or hCaptcha and not set a public/private key in input?
+		$set_captcha_image = false;
+		if(isset($mybb->input['upsetting']['captchaimage'])){
+			$captchaimage = $mybb->input['upsetting']['captchaimage'];
+			$recaptchaprivatekey = $mybb->input['upsetting']['recaptchaprivatekey'];
+			$recaptchapublickey = $mybb->input['upsetting']['recaptchapublickey'];
+			$hcaptchaprivatekey = $mybb->input['upsetting']['hcaptchaprivatekey'];
+			$hcaptchapublickey = $mybb->input['upsetting']['hcaptchapublickey'];
+
+			if(in_array($captchaimage, array(NOCAPTCHA_RECAPTCHA, RECAPTCHA_INVISIBLE)) && (!$recaptchaprivatekey || !$recaptchapublickey)){
+				$set_captcha_image = true;
+			}
+			if(in_array($captchaimage, array(RECAPTCHA_V3)) && (!$recaptchaprivatekey || !$recaptchapublickey || !$recaptchascore)){
+				$set_captcha_image = true;
+			}
+			if(in_array($captchaimage, array(HCAPTCHA, HCAPTCHA_INVISIBLE)) && (!$hcaptchaprivatekey || !$hcaptchapublickey)){
+				$set_captcha_image = true;
+			}
+		}
+
+		//Checking settings for reCAPTCHA or hCaptcha and public/private key not set?
+		$captchaimage = $mybb->settings['captchaimage'];
+		$recaptchaprivatekey = $mybb->settings['recaptchaprivatekey'];
+		$recaptchapublickey = $mybb->settings['recaptchapublickey'];
+		$recaptchascore = $mybb->settings['recaptchascore'];
+		$hcaptchaprivatekey = $mybb->settings['hcaptchaprivatekey'];
+		$hcaptchapublickey = $mybb->settings['hcaptchapublickey'];
+
+		if(in_array($captchaimage, array(NOCAPTCHA_RECAPTCHA, RECAPTCHA_INVISIBLE)) && (!$recaptchaprivatekey || !$recaptchapublickey)){
+			$set_captcha_image = true;
+		}
+		if(in_array($captchaimage, array(RECAPTCHA_V3)) && (!$recaptchaprivatekey || !$recaptchapublickey || !$recaptchascore)){
+			$set_captcha_image = true;
+		}
+		if(in_array($captchaimage, array(6, HCAPTCHA_INVISIBLE)) && (!$hcaptchaprivatekey || !$hcaptchapublickey)){
+			$set_captcha_image = true;
+		}
+		if($set_captcha_image){
+			$mybb->input['upsetting']['captchaimage'] = DEFAULT_CAPTCHA;
 			$lang->success_settings_updated .= $lang->success_settings_updated_captchaimage;
 		}
 
+
 		// If using fulltext then enforce minimum word length given by database
```

Comment:
```suggestion
		}

		// If using fulltext then enforce minimum word length given by database
```

Please undo this extra line insert.